--- aubio-0.4.9/src/io/source_avcodec.c.orig	2024-04-14 23:09:02.187890285 +0100
+++ aubio-0.4.9/src/io/source_avcodec.c	2024-04-14 23:12:44.256674540 +0100
@@ -259,7 +259,7 @@
 
   /* get input specs */
   s->input_samplerate = avCodecCtx->sample_rate;
-  s->input_channels   = avCodecCtx->channels;
+  s->input_channels   = avCodecCtx->ch_layout.nb_channels;
   //AUBIO_DBG("input_samplerate: %d\n", s->input_samplerate);
   //AUBIO_DBG("input_channels: %d\n", s->input_channels);
 
@@ -312,8 +312,10 @@
   // create or reset resampler to/from mono/multi-channel
   if ( s->avr == NULL ) {
     int err;
-    int64_t input_layout = av_get_default_channel_layout(s->input_channels);
-    int64_t output_layout = av_get_default_channel_layout(s->input_channels);
+    AVChannelLayout layout;
+    av_channel_layout_default(&layout,s->input_channels);
+    int64_t input_layout = layout.u.mask;
+    int64_t output_layout = layout.u.mask;
 #ifdef HAVE_AVRESAMPLE
     AVAudioResampleContext *avr = avresample_alloc_context();
 #elif defined(HAVE_SWRESAMPLE)
@@ -369,7 +371,7 @@
   int out_samples = 0;
 #elif defined(HAVE_SWRESAMPLE)
   int in_samples = avFrame->nb_samples;
-  int max_out_samples = AUBIO_AVCODEC_MAX_BUFFER_SIZE / avCodecCtx->channels;
+  int max_out_samples = AUBIO_AVCODEC_MAX_BUFFER_SIZE / avCodecCtx->ch_layout.nb_channels;
   int out_samples = 0;
 #endif /* HAVE_AVRESAMPLE || HAVE_SWRESAMPLE */
   smpl_t *output = s->output;
@@ -436,10 +438,10 @@
   }
 
 #if LIBAVUTIL_VERSION_MAJOR > 52
-  if (avFrame->channels != (sint_t)s->input_channels) {
+  if (avFrame->ch_layout.nb_channels != (sint_t)s->input_channels) {
     AUBIO_WRN ("source_avcodec: trying to read from %d channel(s),"
         "but configured for %d; is '%s' corrupt?\n",
-        avFrame->channels, s->input_channels, s->path);
+        avFrame->ch_layout.nb_channels, s->input_channels, s->path);
     goto beach;
   }
 #else
@@ -448,7 +450,7 @@
 
 #ifdef HAVE_AVRESAMPLE
   in_linesize = 0;
-  av_samples_get_buffer_size(&in_linesize, avCodecCtx->channels,
+  av_samples_get_buffer_size(&in_linesize, avCodecCtx->ch_layout.nb_channels,
       avFrame->nb_samples, avCodecCtx->sample_fmt, 1);
   in_samples = avFrame->nb_samples;
   out_linesize = 0;
@@ -458,7 +460,7 @@
         (uint8_t **)avFrame->data, in_linesize, in_samples);
 #elif defined(HAVE_SWRESAMPLE)
   in_samples = avFrame->nb_samples;
-  max_out_samples = AUBIO_AVCODEC_MAX_BUFFER_SIZE / avCodecCtx->channels;
+  max_out_samples = AUBIO_AVCODEC_MAX_BUFFER_SIZE / avCodecCtx->ch_layout.nb_channels;
   out_samples = swr_convert( avr,
       (uint8_t **)&output, max_out_samples,
       (const uint8_t **)avFrame->data, in_samples);
